localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -410,10 +402,8 @@ void NDTScanMatcher::callback_initial_pose(
         << ". Please check the frame_id in the input topic and ensure it is correct.");
   }
 
-  if (use_dynamic_map_loading_) {
-    std::lock_guard<std::mutex> lock(latest_ekf_position_mtx_);
-    latest_ekf_position_ = initial_pose_msg_ptr->pose.pose.position;
-  }
+  std::lock_guard<std::mutex> lock(latest_ekf_position_mtx_);
+  latest_ekf_position_ = initial_pose_msg_ptr->pose.pose.position;
```

Comment:
How about adding a scope to emphasize that we are getting a lock?
I think this will make it less likely to be buggy when code is added behind it in the future.
```suggestion
  {
    std::lock_guard<std::mutex> lock(latest_ekf_position_mtx_);
    latest_ekf_position_ = initial_pose_msg_ptr->pose.pose.position;
  }
```

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/5873/commits/669ce13e6bbc6019d739c86bf5111d076cbaf5a0